app/profile/ProfilePageClient.tsx

```diff
@@ -0,0 +1,23 @@
+"use client";
+import "./profile.scss";
+import { useAppDispatch } from "@/redux/hooks";
+import { updateUserData } from "@/redux/slices/userSlice/userSlice";
+
+export default function ProfilePageClient({ user }) {
+  const dispatch = useAppDispatch();
+  dispatch(updateUserData(user));

```

Comment:
Z jakiego powodu wykonujesz to od razu po załadowaniu komponentu? Jeżeli jest to faktycznie potrzebne, to musi być wewnątrz useEffect aby uniknąć loopa re-renderingów